test/unit/css.js

```diff
@@ -203,6 +203,59 @@ test( "css() explicit and relative values", 29, function() {
 	equal( $elem.css("opacity"), "1", "'+=0.5' on opacity (params)" );
 });
 
+test( "css() non-px relative values (gh-1711)", 17, function() {
+	var cssCurrent,
+		units = {},
+		$parent = jQuery("#nothiddendiv"),
+		$child = jQuery("#nothiddendivchild", $parent),
+		add = function( prop, val, unit ) {
+			var str = ( val < 0 ? "-=" : "+=" ) + Math.abs(val) + unit;
+			$child.css( prop, str );
+			equal(
+				Math.round( parseFloat( $child.css( prop ) ) ),
+				Math.round( cssCurrent += val * units[ prop ][ unit ] ),
+				"'" + str + "' on " + prop
+			);
+		},
+		getUnits = function( prop ) {
+			units[ prop ] = {
+				"px": 1,
+				"em": parseFloat( $child.css( "fontSize", "16px" ).css( "fontSize" )),
+				"pt": parseFloat( $child.css( prop, "100pt" ).css( prop ) ) / 100,
+				"pc": parseFloat( $child.css( prop, "100pc" ).css( prop ) ) / 100,
+				"cm": parseFloat( $child.css( prop, "100cm" ).css( prop ) ) / 100,
+				"mm": parseFloat( $child.css( prop, "100mm" ).css( prop ) ) / 100,
+				"%" : parseFloat( $child.css( prop, "100%"  ).css( prop ) ) / 100
+			};
+		};
+
+	$parent.css({ height: 1, padding: 0, width: 400 });
+	$child.css({ height: 1, padding: 0 });
+
+	getUnits( "width" );
+	cssCurrent = $child.css( "width", "50%" ).width();
```

Comment:
There's no need to pull in code from other modules here; this should use `.css( "width" )` in analogy to the "lineHeight" block below.